### ISSUE TITLE
[Hunted Forever scenario] Relentless Hulk duplicated, can't generate minimized archive

### ISSUE THREAD

**Issue by Terrorforge** [CleverRaven/Cataclysm-DDA#75072]
### Describe the bug

Ran into the relentless hulk (Mindhunter variant, because I'm running MoM) right in front of a big group of Mi-go. They fought, and the hulk lost badly. When it died, the log said "The mindhunter dies! x2":
![image](https://github.com/user-attachments/assets/304981e6-580d-4b21-94c7-4db72fe37c4b)

This appears to have triggered the mission twice, causing two new Mindhunters to spawn, because I ran into both of them as I was driving around the area.

Then when I was trying to submit this bug report, I was unable to generate a minimized save archive. It would fail with this error, overlapping some other text box I can't read:
![image](https://github.com/user-attachments/assets/328e07a2-8711-4d39-94a7-36a557585c50)

I'm using Catapult, but I tried moving the save to a clean install and nothing changed. Thankfully this was very early in the run so the save folder was small enough that I could just compress and upload the whole thing.

My hunch is that this save weirdness is due to the Nemesis mission being duplicated, but that's just speculation.

### Attach save file

[New Mind.zip](https://github.com/user-attachments/files/16271101/New.Mind.zip)
As mentioned, this is not a compressed save archive generated through the in-game debug utility, it's just the whoel save folder compressed into a .zip

### Steps to reproduce

Unclear. Load the provided save for an example. I've marked the location of the hulks with cyan M's on the map, and provided screenshots of their map locations.

### Expected behavior

There's supposed to be exactly one hulk that pursues you relentless, not one that reproduces my mitosis.

### Screenshots

![image](https://github.com/user-attachments/assets/ebacfbf9-6dae-4dc4-b2e9-3ae72e37e7af)
![image](https://github.com/user-attachments/assets/01986c6e-cbcd-4fbc-a091-545e41ec7e6f)
This guy is in this location on the map.

![image](https://github.com/user-attachments/assets/0b63128a-b31b-47d0-b0b9-bc4c4cd77f0e)
![image](https://github.com/user-attachments/assets/92b12bb5-d5bc-4d5f-a1e3-f564f8229eeb)
And there's another one over here, some 40-odd tiles to the west.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.4651 (22H2)
- Game Version: f830123 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mind Over Matter [mindovermatter],
    Bionic Slots [cbm_slots],
    Stats Through Kills [stats_through_kills]
]

### Additional context

This could be a feature, though. Adding additional hulks at certain upgrade levels would certainly increase the difficulty in a more exciting way than just making it 1% faster, but obviouly it would need to happen in a controlled manner.

**Comment by harakka** [comment]
Couldn't repro this on  latest CI Linux tiles build, but may be OS related anyway.

- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 24.04 LTS; Release: 24.04; Codename: noble; 
- Game Version: cdda-experimental-2024-06-24-0752 1b599f0 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Mind Over Matter [mindovermatter],
    Bionic Slots [cbm_slots],
    Stats Through Kills [stats_through_kills]
]

**Comment by Terrorforge** [comment]
Root bug may be a duplicate of #74621 because there was mention of multiple hulks, but the issue seems to have focused on whether the hulk should be respawning at all.

**Comment by Terrorforge** [comment]
I tried debug killing one of the duplicate Nemeses. I got the death message twice again, but using debug "display hordes" I could confirm that this spawned not two new Nemesis mongroups, but _at least twelve of them_, which is presumably what's ultimately responsible for the ridiculous numbers of nemeses seen in #74669.

Tangentially @PatrikLundell the spawning code for the nemesis is chiefly in mission_start.cpp (line 61, mission_start::kill_nemesis), spreading out into tuff like the add_nemesis function in overmapbuffer.cpp and place_nemesis in overmap.cpp

Also paging @RenechCDDA. Your scenario is pretty severely borked.

**Comment by PatrikLundell** [comment]
The reason for the multiplication seems to reside in mission.cpp, mission::on_creature_death.
The logic is a bit flawed, so one horde is added for each mission in the array, rather than once.
The double death message is probably caused by whatever causes double death messages for zombie children, but doesn't affect the number of hordes spawned, as that code is called only once.

It can be noted that the pointers to the relevant code made identification of the problem a fairly easy task.

**Comment by Terrorforge** [comment]
I thought that might be the case, but I couldn't quite untangle the loops to figure out if and where the problem was. Thanks for fixing it.

**Comment by PatrikLundell** [comment]
All the code was actually there. I just moved existing lines around a little...
